go/cmd/vtcombo/main.go

```diff
@@ -119,9 +121,19 @@ func main() {
 
 	// parse the input topology
 	tpb := &vttestpb.VTTestTopology{}
-	if err := prototext.Unmarshal([]byte(*protoTopo), tpb); err != nil {
-		log.Errorf("cannot parse topology: %v", err)
-		exit.Return(1)
+	switch {
+	case *jsonTopo != "":
+		decoder := json.NewDecoder(strings.NewReader(*jsonTopo))
```

Comment:
The `encoding/json` package is **not** going to decode properly a JSON payload into a protobuf struct like `VTTestTopology`. You need to use https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson instead or we're going to see weird behaviors.

Reply:
TIL! Thanks 🙇 ❤️